test/test_custom_ops.py

```diff
@@ -1675,6 +1675,33 @@ def foo_backward(ctx, saved, grad_out):
         (gx,) = torch.autograd.grad(y, x)
         self.assertEqual(gx, x.cos())
 
+    def test_define(self):
+        lib = self.lib()
+        torch.library.define(f"{self.test_ns}::foo", "(Tensor x) -> Tensor", lib=lib)
+
+        def f(x):
+            return torch.from_numpy(np.sin(x.numpy()))
+
+        lib.impl("foo", f, "CPU")
```

Comment:
Decorator style is supported for impl. Decorator style is not supported for define. This is because there is no function to actually decorate for define.

Reply:
In the next PR up you will see decorator support for impl in the tests.